extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/functionExtractors.scala

```diff
@@ -29,13 +30,17 @@ object FunctionExtractor {
     loadExtractorsToMap[FunctionExtractor]
   }
 
-  def buildFunctionIdentFromQualifiedName(qualifiedName: String): (String, Option[String]) = {
-    val parts: Array[String] = qualifiedName.split("\\.", 2)
-    if (parts.length == 1) {
-      (qualifiedName, None)
+  private[authz] def buildFunctionFromQualifiedName(qualifiedName: String): Function = {
+    val partsLen = if (isSparkV34OrGreater) 3 else 2
+    val parts: Array[String] = qualifiedName.split("\\.", partsLen)
+    val (functionName, database) = if (parts.length == 1) {
+      (None, qualifiedName)
+    } else if (parts.length == 3) {
+      (Some(parts.tail.head), parts.last)
     } else {
-      (parts.last, Some(parts.head))
+      (Some(parts.head), parts.last)
     }
+    Function(functionName, database)
```

Comment:
Seems reversed.

Reply:
> Seems reversed.

I adjust the function field order to kepp consistent with spark.